packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,129 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
         setStyles(alteredStyles);
     }, [props.style, props.aspectRatio, videoAspectRatio]);
 
+    useEffect(() => {
+        if (props.showControls) {
+            showControlsHandler();
+        }
+    }, [fullScreen, props.showControls]);
+
+    function showControlsHandler(): void {
+        clearTimeout(timeoutRef.current as NodeJS.Timeout);
+
+        setShowControls(true);
+        timeoutRef.current = global.setTimeout(() => {
+            setShowControls(false);
+        }, 5000);
+    }
+
+    const onVideoPressHandler = useCallback((): void => {
+        if (!props.showControls) {
+            return;
+        }
+        if (!showControls) {
+            showControlsHandler();
+        } else {
+            clearTimeout(timeoutRef.current as NodeJS.Timeout);
+            setShowControls(false);
+        }
+    }, [props.showControls, showControls]);
+
+    function fullScreenHandler(isFullScreen: boolean): void {
+        setFullScreen(isFullScreen);
+        StatusBar.setHidden(isFullScreen);
+    }
+
+    const videoProps: VideoProperties = {
+        testID: props.name,
+        source: { uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined },
+        muted: props.muted,
+        repeat: props.loop,
+        controls: props.showControls,
+        onLoadStart: () => setStatus(StatusEnum.LOADING),
```

Comment:
I suggest using `useCallback` here and below

Reply:
this is the same case as onLoad